ui/tests/integration/components/open-api-explorer/swagger-ui-test.js

```diff
@@ -8,31 +8,53 @@ import { setupRenderingTest } from 'vault/tests/helpers';
 import { waitUntil, find } from '@ember/test-helpers';
 import { setupEngine } from 'ember-engines/test-support';
 import { setupMirage } from 'ember-cli-mirage/test-support';
-import { render } from '@ember/test-helpers';
+import { render, fillIn } from '@ember/test-helpers';
 import { hbs } from 'ember-cli-htmlbars';
 
+const SELECTORS = {
+  container: '[data-test-swagger-ui]',
+  searchInput: '.operation-filter-input',
+  apiPathBlock: '.opblock-post',
+};
+
 module('Integration | Component | open-api-explorer | swagger-ui', function (hooks) {
   setupRenderingTest(hooks);
   setupEngine(hooks, 'open-api-explorer');
   setupMirage(hooks);
   hooks.beforeEach(function () {
     this.store = this.owner.lookup('service:store');
-  });
 
-  test('it renders', async function (assert) {
-    assert.expect(2);
     const openApiResponse = this.server.create('open-api-explorer');
     this.server.get('sys/internal/specs/openapi', () => {
       return openApiResponse;
     });
 
-    await render(hbs`<SwaggerUi/>`, {
-      owner: this.engine,
-    });
+    this.totalApiPaths = Object.keys(openApiResponse.paths).length;
+
+    this.renderComponent = async () => {
+      await render(hbs`<SwaggerUi/>`, {
+        owner: this.engine,
+      });
+    };
+  });
+
+  test('it renders', async function (assert) {
+    await this.renderComponent();
+
+    await waitUntil(() => find(SELECTORS.container));
+
+    assert.dom(SELECTORS.container).exists('renders component');
+    assert.dom(SELECTORS.apiPathBlock).exists({ count: this.totalApiPaths }, 'renders all api paths');
+  });
+
+  test('it can search', async function (assert) {
+    await this.renderComponent();
+
+    await waitUntil(() => find(SELECTORS.searchInput));
+    await fillIn(SELECTORS.searchInput, 'token');
 
-    await waitUntil(() => find('[data-test-swagger-ui]'));
-    assert.dom('[data-test-swagger-ui]').exists('renders component');
-    await waitUntil(() => find('.operation-filter-input'));
-    assert.dom('.opblock-post').exists({ count: 2 }, 'renders two blocks');
+    // for some reason search results are not rendered immediately in tests,
```

Comment:
yeah, i tried many variations of `waitUntil`, `waitFor` and `setTimeout` to wait for the search results to update to no avail. what's bizarre is that when i add a `await this.pauseTest()` immediately after `fillIn('some search term')`, the text input is still empty and the search results don't update, but if i insert a `debugger` the search input and results work exactly as expected. so there's something going on with timing here. 

unfortunately since the execution of the `opsFilter` function (aka our custom search function) happens outside of our context (within the source code of `swagger-ui-dist`, it's difficult to debug. my thoughts were exactly what you mentioned -- if our search function errors, the whole page breaks, and so asserting that the search input gets entered correctly tells us that the page is "working" (for the most part). 😕